omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/ModuleActionController1_8Tests.java

```diff
@@ -0,0 +1,162 @@
+		        .getResourceBySupportedClass(ModuleAction.class);
+		moduleActionResource.setModuleFactoryWrapper(mockModuleFactory);
+		
+		ModuleResource1_8 moduleResource = (ModuleResource1_8) restService.getResourceBySupportedClass(Module.class);
+		moduleResource.setModuleFactoryWrapper(mockModuleFactory);
+	}
+	
+	@Test
+	public void shouldStartAtlasModule() throws Exception {
+		//sanity check
+		assertThat(mockModuleFactory.startedModules, not(hasItem(atlasModule)));
+		SimpleObject deserialized = deserialize(handle(newPostRequest(getURI(), "{\"action\":\"start\", \"modules\":[\""
+		        + getUuid() + "\"]}")));
+		assertThat(mockModuleFactory.startedModules, hasItem(atlasModule));
+	}
+	
+	@Test
+	public void shouldStopAtlasModule() throws Exception {
+		mockModuleFactory.startedModules.add(atlasModule);
+		SimpleObject deserialized = deserialize(handle(newPostRequest(getURI(), "{\"action\":\"stop\", \"modules\":[\""
+		        + getUuid() + "\"]}")));
+		assertThat(mockModuleFactory.startedModules, not(hasItem(atlasModule)));
+	}
+	
+	@Test
+	public void shouldDoNothingIfAtlasModuleAlreadyStarted() throws Exception {
+		mockModuleFactory.startedModules.add(atlasModule);
+		//sanity check
+		assertThat(mockModuleFactory.startedModules, hasItem(atlasModule));
+		assertThat(mockModuleFactory.loadedModules, hasItem(atlasModule));
+		
+		SimpleObject deserialized = deserialize(handle(newPostRequest(getURI(), "{\"action\":\"start\", \"modules\":[\""
+		        + getUuid() + "\"]}")));
+		
+		//check if state preserved
+		assertThat(mockModuleFactory.startedModules, hasItem(atlasModule));
+		assertThat(mockModuleFactory.loadedModules, hasItem(atlasModule));
+	}
+	
+	@Test
+	public void shouldDoNothingIfAtlasModuleAlreadyStopped() throws Exception {
+		//sanity check
+		assertThat(mockModuleFactory.startedModules, not(hasItem(atlasModule)));
+		assertThat(mockModuleFactory.loadedModules, hasItem(atlasModule));
+		
+		SimpleObject deserialized = deserialize(handle(newPostRequest(getURI(), "{\"action\":\"stop\", \"modules\":[\""
+		        + getUuid() + "\"]}")));
+		
+		//check if state preserved
+		assertThat(mockModuleFactory.startedModules, not(hasItem(atlasModule)));
+		assertThat(mockModuleFactory.loadedModules, hasItem(atlasModule));
+	}
+	
+	@Test
+	public void shouldUnloadAtlasModule() throws Exception {
+		//sanity check
+		assertThat(mockModuleFactory.loadedModules, hasItem(atlasModule));
+		
+		SimpleObject deserialized = deserialize(handle(newPostRequest(getURI(), "{\"action\":\"unload\", \"modules\":[\""
+		        + getUuid() + "\"]}")));
+		assertThat(mockModuleFactory.loadedModules, not(hasItem(atlasModule)));
+	}
+	
+	@Test
+	public void shouldStartAllModules() throws Exception {
+		SimpleObject deserialized = deserialize(handle(newPostRequest(getURI(),
+		    "{\"action\":\"start\", \"allModules\":\"true\"}")));
+		
+		for (Module loadedModule : mockModuleFactory.loadedModules) {
```

Comment:
With way how @Before is written, it cannot be other way, but sanity check won't hurt, I've added assertion